PMIXv2x: OPAL_ASSEMBLY_ARCH macro definition missing in v3.0.x and v3.1.x

In compiling HEAD of v3.0.x and v3.1.x, I get a bunch of warnings like this:

```
775 In file included from /home/jsquyres/git/ompi-crossover/opal/mca/pmix/pmix2x/pmix/src/atomics/sys/atomic.h:162,
776                  from /home/jsquyres/git/ompi-crossover/opal/mca/pmix/pmix2x/pmix/src/threads/thread_usage.h:31,
777                  from /home/jsquyres/git/ompi-crossover/opal/mca/pmix/pmix2x/pmix/src/class/pmix_object.h:131,
778                  from /home/jsquyres/git/ompi-crossover/opal/mca/pmix/pmix2x/pmix/src/mca/base/base.h:31,
779                  from pmix_mca_base_component_compare.c:25:
780 /home/jsquyres/git/ompi-crossover/opal/mca/pmix/pmix2x/pmix/src/atomics/sys/gcc_builtin/atomic.h: In function ‘pmix_atomic_rmb’:
781 /home/jsquyres/git/ompi-crossover/opal/mca/pmix/pmix2x/pmix/src/atomics/sys/gcc_builtin/atomic.h:63:5: warning: "OPAL_ASSEMBLY_ARCH" is not defined, evaluates to 0 [-Wundef]
782  #if OPAL_ASSEMBLY_ARCH == OPAL_X86_64
783      ^~~~~~~~~~~~~~~~~~
```

Do we care / do we need to fix this?  Is there an easy fix / commit we can cherry-pick from master and/or pmix to fix this issue?

--- a/opal/mca/pmix/pmix2x/pmix/src/atomics/sys/gcc_builtin/atomic.h
+++ b/opal/mca/pmix/pmix2x/pmix/src/atomics/sys/gcc_builtin/atomic.h
@@ -15,7 +15,7 @@
  *                         reserved.
  * Copyright (c) 2016-2017 Research Organization for Information Science
  *                         and Technology (RIST). All rights reserved.
- * Copyright (c) 2018      Intel, Inc. All rights reserved.
+ * Copyright (c) 2018-2019 Intel, Inc.  All rights reserved.
  * Copyright (c) 2018      Triad National Security, LLC. All rights
  *                         reserved.
  * $COPYRIGHT$
@@ -60,7 +60,7 @@ static inline void pmix_atomic_mb(void)
 
 static inline void pmix_atomic_rmb(void)
 {
-#if OPAL_ASSEMBLY_ARCH == OPAL_X86_64
+#if PMIX_ASSEMBLY_ARCH == PMIX_X86_64
     /* work around a bug in older gcc versions where ACQUIRE seems to get
      * treated as a no-op instead of being equivalent to
      * __asm__ __volatile__("": : :"memory") */

